src/docfx/build/xref/XrefMap.cs

```diff
@@ -56,6 +57,20 @@ public void OutputXrefMap(Context context)
             context.WriteJson(models, "xrefmap.json");
         }
 
+        public static DataTypeAttribute GetNonXrefPropertyAttribute(IEnumerable<DataTypeAttribute> attributes, List<Error> errors, string jsonPath)
+        {
+            try
+            {
+                var otherAttributes = attributes.Where(attr => !(attr is XrefPropertyAttribute));
+                return otherAttributes.SingleOrDefault(attr => !(attr is XrefPropertyAttribute));
+            }
+            catch (InvalidOperationException)
+            {
+                errors.Add(Errors.InvalidMultipleAttributes(jsonPath));
```

Comment:
we only add error code for end users (writers). for errors introduced by us (developers), just let it throw and make docfx crash.

Reply:
Good to know!

Reply:
that means the schema is owned by developer?

---
In reply to: [223567812](https://github.com/dotnet/docfx/pull/3479#discussion_r223567812) [](ancestors = 223567812)

Reply:
yes, there is currently no way for people to extend schemas other than us.

Reply:
In the future we might add the ability to extend docfx with custom schemas, that is a different scenario and requires extending templates.